plugins/plugins.go

```diff
@@ -576,7 +576,7 @@ func (m *Manager) Labels() map[string]string {
 	return m.Config.Labels
 }
 
-// InterQueryBuiltinCacheConfig returns the configuration for the inter-query cache.
+// InterQueryBuiltinCacheConfig returns the configuration for the inter-query, regex builtin and glob builtin cache.
 func (m *Manager) InterQueryBuiltinCacheConfig() *cache.Config {
```

Comment:
This is a misnomer. We could deprecate it and add a new `BuiltinCacheConfig()` method.

Reply:
On a general note: the existing inter-query cache is a by-size cache, and the new glob- and regex caches are by-count. The first is general-purpose, whereas the latter additions are specific to regex and glob entries. I wonder if we should also treat the new caches as general purpose, so that custom built-ins can still use them 🤔 ? I.e. We'd have two caches; one by-size cache for everything that has a calculable size, and one by-count cache for everything that can't be sized.

Reply:
That's an interesting idea. So you have the existing `inter_query_builtin_cache` and you'll have a new `inter_query_builtin_value_cache`. The later would at first be utilized by the `regex` and `blob` builtins. Is this what you had in mind?

Reply:
Pretty much, yeah 👍

Reply:
Updated with new inter-query value cache that builtins can use.